be/src/vec/exec/scan/scan_task_queue.cpp

```diff
@@ -132,6 +205,31 @@ void ScanTaskTaskGroupQueue::update_tg_cpu_share(const taskgroup::TaskGroupInfo&
     }
 }
 
+void ScanTaskTaskGroupQueue::reset_empty_group_entity() {
+    int user_g_cpu_hard_limit = 0;
+    bool contains_empty_group = false;
+    for (auto* entity : _group_entities) {
+        if (!entity->is_empty_group_entity()) {
+            user_g_cpu_hard_limit += entity->cpu_share();
+        } else {
+            contains_empty_group = true;
+        }
+    }
+
+    // 0 <= user_g_cpu_hard_limit <= 100, bound by FE
+    // user_g_cpu_hard_limit = 0 means no group exists
+    int empty_group_cpu_share = 100 - user_g_cpu_hard_limit;
+    if (empty_group_cpu_share > 0 && empty_group_cpu_share < 100 && !contains_empty_group) {
+        _empty_group_entity->update_empty_cpu_share(empty_group_cpu_share);
+        _enqueue_task_group(_empty_group_entity);
+    } else if ((empty_group_cpu_share == 0 || empty_group_cpu_share == 100) &&
+               contains_empty_group) {
+        _dequeue_task_group(_empty_group_entity);
```

Comment:
the op is dangerous，do not set the empty cpu share， so `empty_group_entity` the `cpu_share` is undefine